### PR TITLE
Terraform of ETCD and CT server

### PR DESCRIPTION
**Pull request by AndrewGorton**
Key rotation

### CODE REVIEW

File: terraform/certificate-transparency.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - sudo
+      - docker
+    ssh-authorized-keys:
+      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDcntQbymoWmb4l2xztDMu+6KAa9+ykyarqxFhkejGXNy0fgUlm0P8Td7vI5/AU4YAGiOgLjq38B/OADhv40x0L66Zi5TkHAL+XWgQ+yKPjsc170WCJy6nv5gInPhPZ794H6v0JT+9i1/f7Cs5PiVrWfIF9oj4ZtUpDUjSPzsk86b07n+384thSO6gB5fugCkIYan7Uyk05v8yAoHpkelyNTzLdRWtIqOQvtt1vQnC6ZBV7M6JAQQAzXnJnhTaPjY6Dy7a3co5HQonhjsDSo151dcfngyEmdbs4yRB4ZHBsyno26bSNKzKv2ltKrVr+/NKgtbyIDdUXaopuT9drKvM3 foromsharma@gmail.com
+  - name: philandstuff
+    groups:
+      - sudo
+      - docker
+    ssh-authorized-keys:
+      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCsYvhPkvKma2RwWVwUNJ2fcoNjlObibUka0fl5WcctTokz8RISDkOAPjz+L0QmGsq/Dhgn/mbspKFyNKvQsujeooEpaO+cjYPXoUF1NTxhbUrMEu2uE+CPoj4zPbAZ9A6LdxkL13F3gJwZi9k8K9+6AnJeB4IrCLS5KHRiL+tBZMLPvKnn54M6PCfkWJH6zVO92plFTpRX1W4GanUwYGTemFX0qUwFgZu9OJvGiR+ZQKJMQ2Y3O+IMHv1KusfBrO+Q4iWn+A5VcJRoxml3lr0fYlbaetCpszkaY5HbQsLPDGtvsN1jXj2ghVe2ZbK8SKe13g+LSqYxmIX5qCAoEi+J philip.potter@digital.cabinet-office.gov.uk
```
Comment (philandstuff) [inline terraform/certificate-transparency.yml]: this is my old key.

File: terraform/etcd.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
+    listen-peer-urls: http://$private_ipv4:2380,http://$private_ipv4:7001
+  units:
+    - name: etcd2.service
+      command: start
+    - name: fleet.service
+      command: start
+
+users:
+  - name: andrewg
```
Comment (philandstuff) [inline terraform/etcd.yml]: missing me and slawek?

Comment (philandstuff) [inline terraform/etcd.yml]: looking further: what's the relationship between this file and etcd.yml.template, and why do they both have `users` fields?

File: terraform/etcd.yml.template [hunk 3]
```diff
@@ -25,4 +25,22 @@ users:
...
+      - sudo
+      - docker
+    ssh-authorized-keys:
+      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDcntQbymoWmb4l2xztDMu+6KAa9+ykyarqxFhkejGXNy0fgUlm0P8Td7vI5/AU4YAGiOgLjq38B/OADhv40x0L66Zi5TkHAL+XWgQ+yKPjsc170WCJy6nv5gInPhPZ794H6v0JT+9i1/f7Cs5PiVrWfIF9oj4ZtUpDUjSPzsk86b07n+384thSO6gB5fugCkIYan7Uyk05v8yAoHpkelyNTzLdRWtIqOQvtt1vQnC6ZBV7M6JAQQAzXnJnhTaPjY6Dy7a3co5HQonhjsDSo151dcfngyEmdbs4yRB4ZHBsyno26bSNKzKv2ltKrVr+/NKgtbyIDdUXaopuT9drKvM3 foromsharma@gmail.com
+  - name: philandstuff
+    groups:
+      - sudo
+      - docker
+    ssh-authorized-keys:
+      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCsYvhPkvKma2RwWVwUNJ2fcoNjlObibUka0fl5WcctTokz8RISDkOAPjz+L0QmGsq/Dhgn/mbspKFyNKvQsujeooEpaO+cjYPXoUF1NTxhbUrMEu2uE+CPoj4zPbAZ9A6LdxkL13F3gJwZi9k8K9+6AnJeB4IrCLS5KHRiL+tBZMLPvKnn54M6PCfkWJH6zVO92plFTpRX1W4GanUwYGTemFX0qUwFgZu9OJvGiR+ZQKJMQ2Y3O+IMHv1KusfBrO+Q4iWn+A5VcJRoxml3lr0fYlbaetCpszkaY5HbQsLPDGtvsN1jXj2ghVe2ZbK8SKe13g+LSqYxmIX5qCAoEi+J philip.potter@digital.cabinet-office.gov.uk
```
Comment (philandstuff) [inline terraform/etcd.yml.template]: old key again

**Comment by philandstuff** [comment]
my new key is in user_data.yml on master and ends with `cardno:000603815000`

**Comment by om-sharma** [comment]
my key is also the older one. Please pick the new from `user-data.yaml` file

File: terraform/etcd.yml (hunk 2 above)
Comment (AndrewGorton) [inline terraform/etcd.yml]: @philandstuff: etcd.yml is generated from etcd.yml.template via the generate_etcd_cluster_token.sh script.

**Pull request merged**